hazelcast/src/main/java/com/hazelcast/map/impl/tx/TransactionalMapProxy.java

```diff
@@ -315,12 +319,9 @@ public void delete(Object key) {
     @SuppressWarnings("unchecked")
     public Set keySet(Predicate predicate) {
         checkTransactionState();
-        if (predicate == null) {
-            throw new NullPointerException("Predicate should not be null!");
-        }
-        if (predicate instanceof PagingPredicate) {
-            throw new NullPointerException("Paging is not supported for Transactional queries!");
-        }
+        checkNotNull(predicate, "Predicate should not be null!");
+        checkNotInstanceOf(PagingPredicate.class, predicate, "Paging is not supported for Transactional queries!");
```

Comment:
This method throws an `IllegalArgumentException` instead of `NullPointerException` before. But this is not considered as API change, isn't it? Maybe a map test will fail, but we'll see :)